openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/AppRepository.java

```diff
@@ -199,6 +307,57 @@ public ResultList<AppRunRecord> listAppRuns(UUID appId, int limitParam, int offs
     }
   }
 
+  public <T> ResultList<T> listAllAppExtension(
+      App app, boolean byName, Class<T> clazz, AppExtension.ExtensionType extensionType) {
+    return listAppExtension(app, byName, Integer.MAX_VALUE, 0, clazz, extensionType);
+  }
+
+  public <T> T getLatestExtension(
+      App app, boolean byName, Class<T> clazz, AppExtension.ExtensionType extensionType) {
+    List<String> result;
+    if (byName) {
```

Comment:
@pmbrull we should spearate these into different funct getlatestExtensionByName and getlatestExtensionById

Reply:
wanted to avoid copy-pasting the same method, but fair enough, on it.